Parse/src/main/java/com/parse/ParseLogInterceptor.java

```diff
@@ -93,67 +109,60 @@ public void writeLine(String str) {
   }
 
   private static class ProxyInputStream extends InputStream {
+    // Helper stream to proxy the original input stream to other input stream
+    InputStream originalInput;
+    PipedInputStream proxyInput;
+    PipedOutputStream proxyOutput;
```

Comment:
For `proxyInput` and `proxyOutput`, since we use them in a try/catch block, it is not so easy to make them final, so I will make them private and make `originalInput` private and final.